controllers/reconcilers/kserve_metrics_dashboard_reconciler.go

```diff
@@ -0,0 +1,202 @@
+	kservev1beta1 "github.com/kserve/kserve/pkg/apis/serving/v1beta1"
+	"github.com/opendatahub-io/odh-model-controller/controllers/comparators"
+	"github.com/opendatahub-io/odh-model-controller/controllers/processors"
+	"github.com/opendatahub-io/odh-model-controller/controllers/resources"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Query struct {
+	Title string `json:"title"`
+	Query string `json:"query"`
+}
+
+type Config struct {
+	Title   string  `json:"title"`
+	Type    string  `json:"type"`
+	Queries []Query `json:"queries"`
+}
+
+type MetricsDashboardConfigMapData struct {
+	Data []Config `json:"data"`
+}
+
+var _ SubResourceReconciler = (*KserveMetricsDashboardReconciler)(nil)
+var ovmsData []byte
+var tgisData []byte
+
+// var vllmData []byte
+
+type KserveMetricsDashboardReconciler struct {
+	NoResourceRemoval
+	client           client.Client
+	telemetryHandler resources.ConfigMapHandler
+	deltaProcessor   processors.DeltaProcessor
+}
+
+func NewKserveMetricsDashboardReconciler(client client.Client) *KserveMetricsDashboardReconciler {
+	return &KserveMetricsDashboardReconciler{
+		client:           client,
+		telemetryHandler: resources.NewConfigMapHandler(client),
+		deltaProcessor:   processors.NewDeltaProcessor(),
+	}
+}
+
+func (r *KserveMetricsDashboardReconciler) Reconcile(ctx context.Context, log logr.Logger, isvc *kservev1beta1.InferenceService) error {
+
+	// Create Desired resource
+	desiredResource, err := r.createDesiredResource(ctx, log, isvc)
+	if err != nil {
+		return err
+	}
+
+	// Get Existing resource
+	existingResource, err := r.getExistingResource(ctx, log, isvc)
+	if err != nil {
+		return err
+	}
+
+	// Process Delta
+	if err = r.processDelta(ctx, log, desiredResource, existingResource); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *KserveMetricsDashboardReconciler) createDesiredResource(ctx context.Context, log logr.Logger, isvc *kservev1beta1.InferenceService) (*corev1.ConfigMap, error) {
+
+	// resolve SR
+	isvcRuntime := isvc.Spec.Predictor.Model.Runtime
+	runtime := &kservev1alpha1.ServingRuntime{}
+	if err := r.client.Get(ctx, types.NamespacedName{Name: *isvcRuntime, Namespace: isvc.Namespace}, runtime); err != nil {
+		log.Error(err, "Could not determine servingruntime for isvc")
+	}
+
+	servingRuntimeImage := runtime.Spec.Containers[0].Image
+	re := regexp.MustCompile(`/([^/@]+)[@:]`)
+	findImageName := re.FindStringSubmatch(servingRuntimeImage)
+	servingRuntime := findImageName[1]
+
+	var data []byte
+	switch servingRuntime {
+	case "openvino_model_server":
+		if ovmsData == nil {
+			ovmsData, err := os.ReadFile("ovms-metrics.json")
+			if err != nil {
+				log.Error(err, "Unable to load metrics dashboard template file:", err)
```

Comment:
if there are any failures to load the data, is it safe to continue the execution?
because the data will be empty and the configMap will be created with no data on it.